### PR TITLE
Implement inverse trigonometric functions.

### PR DESCRIPTION
**Pull request by takagi**
Merge #1534 first.

This PR implements inverse trigonometric functions: `arcsin`, `arccos` and `arctan`.

### CODE REVIEW

**Comment by okuta** [comment]
Please split PR trigonometric functions and `unary_math_function_test`.

**Comment by takagi** [comment]
It is already split. The commit list includes `unary_math_function_test`'s commits because the PR is compared to the current `master`. After #1534 `unary_math_function_test` will be merged, the list will contain only commits for inverse trigonometric functions.

Before that, please use the following comparison for reviewing.
https://github.com/takagi/chainer/compare/unary-function-test...takagi:inverse-trigonometric

**Comment by takagi** [comment]
The commit list is now cleaned.

File: chainer/functions/math/trigonometric.py [hunk 1]
```diff
@@ -100,3 +100,143 @@ def backward(self, x, gy):
...
+        numpy.sqrt(gx, out=gx)
+        numpy.reciprocal(gx, out=gx)
+        numpy.negative(gx, out=gx)
+        gx *= gy[0]
+        return gx,
+
+    def backward_gpu(self, x, gy):
+        gx = cuda.elementwise(
+            'T x, T gy', 'T gx',
+            'gx = (T)-1.0 / sqrt((T)1.0 - x * x)',
```
Comment (unnonouno) [inline chainer/functions/math/trigonometric.py]: How about using rsqrt?

File: chainer/functions/math/trigonometric.py [hunk 2]
```diff
@@ -100,3 +100,143 @@ def backward(self, x, gy):
...
+        gx += 1
+        numpy.sqrt(gx, out=gx)
+        numpy.reciprocal(gx, out=gx)
+        gx *= gy[0]
+        return gx,
+
+    def backward_gpu(self, x, gy):
+        gx = cuda.elementwise(
+            'T x, T gy', 'T gx',
+            'gx = (T)1.0 / sqrt((T)1.0 - x * x)',
```
Comment (unnonouno) [inline chainer/functions/math/trigonometric.py]: How about using rsqrt?

File: chainer/functions/math/trigonometric.py [hunk 3]
```diff
@@ -100,3 +100,143 @@ def backward(self, x, gy):
...
+            'arcsin_bwd'
+        )(x[0], gy[0])
+        return gx,
+
+
+def arcsin(x):
+    """Elementwise arcsine function.
+
+    .. math::
+       y_i = \\arcsin x_i.
```
Comment (unnonouno) [inline chainer/functions/math/trigonometric.py]: ``\\arcsin(x_i)``?

File: chainer/functions/math/trigonometric.py [hunk 4]
```diff
@@ -100,3 +100,143 @@ def backward(self, x, gy):
...
+            'arccos_bwd'
+        )(x[0], gy[0])
+        return gx,
+
+
+def arccos(x):
+    """Elementwise arccosine function.
+
+    .. math::
+       y_i = \\arccos x_i.
```
Comment (unnonouno) [inline chainer/functions/math/trigonometric.py]: ``\\arccos(x_i)``?

File: chainer/functions/math/trigonometric.py [hunk 5]
```diff
@@ -100,3 +100,143 @@ def backward(self, x, gy):
...
+            'arctan_bwd'
+        )(x[0], gy[0])
+        return gx,
+
+
+def arctan(x):
+    """Elementwise arctangent function.
+
+    .. math::
+       y_i = \\arctan x_i.
```
Comment (unnonouno) [inline chainer/functions/math/trigonometric.py]: ``\\arctan(x_i)``

File: chainer/functions/math/trigonometric.py (hunk 3 above)
Comment (takagi) [inline chainer/functions/math/trigonometric.py]: I suppose the mathematical form usually do not come with parenthesis `y = arcsin x` and the current line is rendered correctly, how about it?

File: chainer/functions/math/trigonometric.py (hunk 1 above)
Comment (takagi) [inline chainer/functions/math/trigonometric.py]: Fixed!

File: chainer/functions/math/trigonometric.py (hunk 4 above)
Comment (takagi) [inline chainer/functions/math/trigonometric.py]: ditto

File: chainer/functions/math/trigonometric.py (hunk 5 above)
Comment (takagi) [inline chainer/functions/math/trigonometric.py]: ditto

File: chainer/functions/math/trigonometric.py (hunk 3 above)
Comment (unnonouno) [inline chainer/functions/math/trigonometric.py]: OK. I understood it.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**